File: app/src/main/java/com/abrsoftware/myapplication/event/PostEvent.java
package com.abrsoftware.myapplication.event;

import com.abrsoftware.myapplication.helper.EventbusHelper;

public class PostEvent {
    private final GeneralEvent generalEvent;
    private final EventbusHelper eventBus;

    public PostEvent() {
        this.eventBus = EventbusHelper.getInstance();
        this.generalEvent = new GeneralEvent();
    }

    public void postEvent(int type) {
        generalEvent.setTypeEvent(type);
        eventBus.post(generalEvent);
    }

    public <T> void postEvent(int type, T statusResponce) {
        generalEvent.setTypeEvent(type);
        if(statusResponce != null){
            generalEvent.setResponce(statusResponce);
            /*
            if(statusResponce instanceof RecommendationResponce){
                generalEvent.setRecommendationRes(statusResponce);
            }*/
        }
        eventBus.post(generalEvent);
    }


}

File: app/src/main/java/com/abrsoftware/myapplication/event/PostEvent.kt
package com.abrsoftware.myapplication.event

import com.abrsoftware.myapplication.event.GeneralEvent
import com.abrsoftware.myapplication.helper.EventbusHelper

open class PostEvent {
    private val generalEvent: GeneralEvent
    private val eventBus: EventbusHelper

    fun postEvent(type: Int) {
        generalEvent.typeEvent = type
        eventBus.post(generalEvent)
    }

    fun <T> postEvent(type: Int, statusResponce: T?) {
        generalEvent.typeEvent = type
        if (statusResponce != null) {
            generalEvent.responce = statusResponce
        }
        eventBus.post(generalEvent)
    }

    init {
        eventBus = EventbusHelper.SingletonHolder.INSTANCE
        generalEvent = GeneralEvent()
    }
}